cerberus-web/src/main/java/com/nike/cerberus/service/AuthTokenService.java

```diff
@@ -75,45 +99,124 @@ public CerberusAuthToken generateToken(
     checkArgument(StringUtils.isNotBlank(principal), "The principal must be set and not empty");
 
     String id = uuidSupplier.get();
-    String token = authTokenGenerator.generateSecureToken();
     OffsetDateTime now = dateTimeSupplier.get();
 
-    AuthTokenRecord tokenRecord =
-        new AuthTokenRecord()
+    switch (tokenFlag.getIssueType()) {
+      case JWT:
+        try {
+          return getCerberusAuthTokenFromJwt(
+              principal, principalType, isAdmin, groups, ttlInMinutes, refreshCount, id, now);
+        } catch (ApiException e) {
+          String tooLongName = DefaultApiError.AUTH_TOKEN_TOO_LONG.getName();
+          boolean isTooLong =
+              e.getApiErrors().stream().anyMatch(err -> err.getName().equals(tooLongName));
```

Comment:
This is a bit fragile and not type safe. If these lines are going to be removed later then that's fine. Otherwise, consider changing JwtService.java:110 into throwing a checked exception.